packages/venia-ui/lib/components/SavedPaymentsPage/savedPaymentsPage.js

```diff
@@ -0,0 +1,63 @@
+import React, { useMemo } from 'react';
+import { PlusSquare } from 'react-feather';
+
+import { useSavedPaymentsPage } from '@magento/peregrine/lib/talons/SavedPaymentsPage/useSavedPaymentsPage';
+import { mergeClasses } from '@magento/venia-ui/lib/classify';
+
+import Icon from '../Icon';
+import LinkButton from '../LinkButton';
+import { Title } from '../Head';
+
+import { GET_SAVED_PAYMENTS_QUERY } from './savedPaymentsPage.gql';
+import defaultClasses from './savedPaymentsPage.css';
+
+const SavedPaymentsPage = props => {
+    const talonProps = useSavedPaymentsPage({
+        queries: {
+            getSavedPaymentsQuery: GET_SAVED_PAYMENTS_QUERY
+        }
+    });
+
+    const { handleAddPayment, savedPayments, strings } = talonProps;
+
+    const classes = mergeClasses(defaultClasses, props.classes);
+
+    const savedPaymentElements = useMemo(() => {
+        return savedPayments.map(
+            ({ details, public_hash, payment_method_code }) => (
+                // TODO: Clean up in PWA-636
+                <div key={public_hash}>
+                    <div>{payment_method_code}</div>
+                    <div>{JSON.stringify(details, null, 2)}</div>
+                </div>
+            )
+        );
+    }, [savedPayments]);
+
+    return (
+        <div className={classes.root}>
+            <Title>{strings.title}</Title>
+            <h1 className={classes.heading}>{strings.title}</h1>
+            <h3 className={classes.subHeading}>{strings.subHeading}</h3>
```

Comment:
I don't like this pattern anymore 😅 .

It feels like unnecessary overhead for the talon to care about translations.

I think it'd be much cleaner to let the components use `FormattedMessage` and declare the `id` that they want.

I'm trying to remember the arguments against this. One that I remember is that `FormattedMessage` returns a `Fragment` - not a single element - so our grids may get messed up. In that case I propose we create our own `VeniaFormattedMessage` or similar that composes `FormattedMessage` and just wraps it in a `<span>`.

Reply:
> One that I remember is that FormattedMessage returns a Fragment - not a single element - so our grids may get messed up. 

You can provide a `tagName` prop to have it use a different tag like `div`.

Reply:
reverting the change for returning strings object.